### PR TITLE
merge: feat/nanmedian

### PR DESCRIPTION
**Pull request by s3m3dov**
Hey there. 
This PR is related to the subtask https://github.com/unifyai/ivy/issues/14117 from https://github.com/unifyai/ivy/issues/13.

Summary:
1. Added `nanmedian` method to the pytorch frontend, and included test.
2. Updated `median` method of the pytorch frontend because it was behaving differently than `pytorch.median` (like `nanmedian`)

Other related issues:
- https://github.com/unifyai/ivy/issues/13211
- https://github.com/unifyai/ivy/issues/13481

Let's say:
```
>>> a = torch.tensor([1, float('nan'), 3, 2])
>>> b = torch.tensor([[2, 3, 1], [float('nan'), 1, float('nan')]])
```

Expected behavior:
```
>>> a.median()
tensor(nan)
>>> a.nanmedian()
tensor(2.)
>>> b.median(0)
torch.return_types.median(values=tensor([nan, 1., nan]), indices=tensor([1, 1, 1]))
>>> b.nanmedian(0)
torch.return_types.nanmedian(values=tensor([2., 1., 1.]), indices=tensor([0, 1, 0]))
```

Actual behavior for `median` in pytorch frontend:
```
>>> a.median()
tensor(2.)
>>> b.median(0)
torch.return_types.nanmedian(values=tensor([2., 1., 1.]), indices=tensor([0, 1, 0]))
```

This PR aims to change the **actual** behavior to the **expected** behavior.

Reference:
https://pytorch.org/docs/stable/generated/torch.nanmedian.html

Close https://github.com/unifyai/ivy/issues/14117

### CODE REVIEW

**Comment by s3m3dov** [comment]
Close #14117

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by s3m3dov** [comment]
@ivy-seed My PR is waiting for review

**Comment by s3m3dov** [comment]
Hi @RickSanchezStoic, could u please review this PR?
Thank you.

**Comment by s3m3dov** [comment]
Hi, @sreegandh804. Could you please review my PR or assign it to someone?
Thank you.

**Comment by s3m3dov** [comment]
No worries. Of course, I will review and get back to you.

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @s3m3dov 

I tried to run your tests locally and I'm unable to trace back to where the issue is would you take your time and look into this😊

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
